aten/src/ATen/core/typeid.h

```diff
@@ -461,18 +461,18 @@ inline bool operator!=(const TypeMeta& lhs, const TypeMeta& rhs) noexcept {
       _makeTypeMetaDataInstance<T>();                                     \
   }                                                                       \
   template<>                                                              \
-  EXPORT_IF_NOT_GCC const detail::TypeMetaData* TypeMeta::_typeMetaDataInstance<T>() noexcept {     \
+  CAFFE2_API const detail::TypeMetaData* TypeMeta::_typeMetaDataInstance<T>() noexcept {     \
     return &MACRO_CONCAT(detail::_typeMetaDataInstance_, Counter);        \
   }
 #define CAFFE_KNOWN_TYPE(T)                                               \
   template <>                                                             \
-  EXPORT_IF_NOT_GCC TypeIdentifier TypeIdentifier::Get<T>() {             \
+  CAFFE2_API TypeIdentifier TypeIdentifier::Get<T>() {                    \
```

Comment:
Seems that when we are doing explicit instantiation, we should do C10_EXPORT always. In the TypeIdentifier class then, we will need to do 

template <typename T>
C10_IMPORT static TypeIdentifier Get();

Given the situation, one might even need to make the whole TypeIdentifier class declaration marked as C10_IMPORT.